packages/core/services/FileService/DatabaseFileService/test/DatabaseFileService.test.ts

```diff
@@ -100,4 +101,65 @@ describe("DatabaseFileService", () => {
             expect(count).to.equal(6);
         });
     });
+
+    describe("applySelectionFilters", () => {
+        // Setup
+        let databaseFileService: DatabaseFileService;
+        let sqlBuilder: SQLBuilder;
+
+        beforeEach(() => {
+            databaseFileService = new DatabaseFileService({
+                dataSourceNames: ["mock_source"],
+                databaseService,
+                downloadService: new FileDownloadServiceNoop(),
+            });
+
+            sqlBuilder = new SQLBuilder().select("*").from("mock_source");
+        });
+
+        // the sql we produce has new lines that mess up comparison
+        function normalizeSQL(sql: string): string {
+            return sql.replace(/\s+/g, " ").trim();
+        }
+
+        it("correctly modifies SQLBuilder for single index selections (CTRL selection)", () => {
+            // Arrange
+            const selections = [
+                {
+                    indexRanges: [
+                        { start: 0, end: 0 },
+                        { start: 2, end: 2 },
```

Comment:
would it be worthwhile to have a test case for 
                    indexRanges: [
                        { start: 0, end: 0 },
                        { start: 2, end: 6 },
                        ]
 mixing the single and the ranges?  Probably redundant.

Reply:
The original bug was only failing with single range selections such as { start: 0, end: 0 }. I think there is some value in splitting up the tests so that each one covers a different case here.